### ISSUE TITLE
Set a simplified geometry for areas during migration

### ISSUE THREAD

**Issue by asaunier** [c2corg/v6_api#489]
Simplified area geometries would be useful to show the actual area shapes in the area advanced search page. As for now the area advanced search page has no map filter because we cannot use points (as for WP, routes or outings) and because the actual geometries are often too big. See https://github.com/c2corg/v6_api/issues/419

A simplified geometry would probably be relevant for detail view pages as well.

It would make sense that areas have both a "full" geometry (moved to ``geom_detail``? only used for geo-associations? also in the editing form map!?) and a simplified geometry (``geom``? used for advanced search and detail view pages?).

* [ ] Setting this while running the migration script would be relevant.
* [ ] The simplified geometry should be updated whenever the full geometry is modified.

**Comment by tsauerwein** [comment]
> Simplified area geometries would be useful to show the actual area shapes in the area advanced search page.

Even with simplified area geometries, you still have the problem with overlapping areas. E.g. the polygon for France would hide all the other areas.

**Comment by asaunier** [comment]
Indeed. The geometry is only a part of the advanced search problem.

It will most likely not be solved before going live but I think it's worth anticipating it because doing the geometry simplifying during the migration would probably be much easier than to do it later.

**Comment by tsauerwein** [comment]
> It will most likely not be solved before going live but I think it's worth anticipating it because doing the geometry simplifying during the migration would probably be much easier than to do it later.

Not really. Even after the migration you can still add a new column and fill this column with a simplified geometry (with Alembic).

**Comment by asaunier** [comment]
But then you have to create a new script, set alambic up etc.
Doing it in the migration script would probably require less lines of code.

**Comment by tsauerwein** [comment]
We have to set up Alembic anyway, see https://github.com/c2corg/v6_api/issues/476

> Doing it in the migration script would probably require less lines of code.

Not really. If it's run as part of an Alembic migration or the main migration does not change much.

**Comment by stef74** [comment]
do you think it is possible to fix before go live ?

**Comment by tsauerwein** [comment]
As discussed with @asaunier, for the Go-Live we will:

- During the migration simplify the geometries for outings and routes.
- If a new geometry is set on an outing or route, store a simplified geometry.

For now, we will not handle areas.

**Comment by stef74** [comment]
> For now, we will not handle areas.
Probably open dedicated issue to close this one when is done and affect the new one to 6.x milestone

**Comment by tsauerwein** [comment]
Ok, I created a new issue: https://github.com/c2corg/v6_api/issues/529
I am keeping this one for the areas.

**Comment by asaunier** [comment]
@stef74 @desnoes Can this issue be postponed to a later milestone? It doesn't look that critical.

**Comment by desnoes** [comment]
It is fine for me to postpone this issue.
